src/BtLineEdit.cpp

```diff
@@ -57,6 +57,7 @@ void BtLineEdit::lineChanged(Unit::unitDisplay oldUnit, Unit::unitScale oldScale
    QString amt;
    bool force = Brewtarget::hasUnits(text());
    bool ok = false;
+   bool wasChanged = sender() == this;
```

Comment:
tl;dr is that yes, it does :)

This slot is called in two possible ways: somebody doing a unit/scale change on a BtLabel, or the BtLineEdit generating an editingFinished signal. The first is connected when constructing screens, the second is done in the constructor for a BtLineEdit.

If the signal is generated by the BtLineEdit it must be because the field was modified by the user. Mostly. See, editingFinished is also generated every time the field loses focus. We actually end up generating a *lot* of editingFinished signals. So the test about four lines further down makes sure something actually changed. If nothing has changed, we just return. This cuts down on a lot of database interactions.

Since we only want to generate the textModified signal when text has actually been modified, the wasChanged flag and the test just a bit further down means the signal will only be emitted if something really changed in the input field.